micromanager_gui/main_window.py

```diff
@@ -148,92 +156,105 @@ def add_frame_explorer(self, name, array):
         except KeyError:
             self.viewer.add_image(array, name=layer_name)
 
-    # TO DO: add the file name form the save box
-    def _on_mda_frame(self, image, event):
+  
+    def _on_mda_frame(self, image: np.ndarray, event: useq.MDAEvent):
 
-        sequence = event.sequence
+        seq = event.sequence
 
-        event.index.setdefault('t', 0)
-        event.index.setdefault('z', 0)
+        #create temp folder if it doesnt exist using uid
+        tmp_uid_name = Path('tmp_folder') / str(seq.uid)
+        temp_folder = Path(__file__).parent.parent / tmp_uid_name
```

Comment:
don't create temp folders yourself... use [`tempfile.TemporaryDirectory`](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory)

Reply:
but if we need to access the temp folder isn't it easier to have it in the plugin path?

Reply:
not sure I'm following the reasoning... You can still access a TemporaryDirectory using its `.name` attribute (read the docs I linked above)

Reply:
The problem that I have is that the name of the temp folder keeps changing every time _on_mda_frame() is called.

Even if I specify prefix and suffix :
`temp_dir = tempfile.TemporaryDirectory(None, str(seq.uid))`
`print(temp_dir.name)` -> ../44e3f048-bbeb-4bc9-88b8-a1cdc6903dc7**8b44f8jh** 
where 8b44f8jh is added and change every time.

is doing something like that that wrong:

```
try:
    temp_folder = Path(tempfile.gettempdir()), ('uid_' + str(seq.uid))
    Path(temp_folder).mkdir(parents=True, exist_ok=False)
except FileExistsError:
    pass
```

Reply:
just like temp_dir deletion shouldn't be in `_on_mda_frame`, nor should tempdir creation... you should create the tempdir _once_ ... on the `onMDAStarted` started signal.  (just think of `_on_mda_frame` as "stuff that needs to be done on _every_ frame... if something you're trying to do doesn't need to happen on every frame, it doesn't belong in `_on_mda_frame`).  create new methods if you need to

Reply:
Thanks Talley, this helped me a lot! I moved everything out of _on_mda_frame